Configuration module

luciehopp opened the pull request.

PascalSchroederDE committed:
Delete unrefactored analyzer

Commit:
Merge pull request #4 from NoxAG/algorithmrefactoring

Delete unrefactored analyzer

luciehopp committed:
Merge branch 'master' of https://github.com/NoxAG/NewNox-Application.git into ConfigurationModule

luciehopp committed:
Run Button und Textfield für das Configuration Module

luciehopp committed:
Nur noch Buttons, kein Textfield. 
Methode getRunButton(), die den Run Button übergibt.

luciehopp committed:
Methoden getOpenButton() und getFileChooser() hinzugefügt.

tobiasxschmidt commented on src/main/java/com/noxag/newnox/ui/configurationmodule/ConfigurationPane.java:
```diff
@@ -1,28 +1,71 @@
+
+public class ConfigurationPane extends BorderPane {
+
+    public Button btnrun, btnopen;
```
hier sollte private statt public genutzt werden

Likedeel replied:
is das buttons wirklich notwendig (ernstgemeinte Frage )

Likedeel replied:
also dass es private is

tobiasxschmidt commented on src/main/java/com/noxag/newnox/ui/configurationmodule/ConfigurationPane.java:
```diff
@@ -1,28 +1,71 @@
+public class ConfigurationPane extends BorderPane {
+
+    public Button btnrun, btnopen;
+    public FileChooser fileChooser;
```
hier auch private

tobiasxschmidt commented on src/main/java/com/noxag/newnox/ui/configurationmodule/ConfigurationPane.java:
```diff
@@ -1,28 +1,71 @@
+
+    public ConfigurationPane() {
+
+        ConfigurationTabPane configTabPane = new ConfigurationTabPane();
```
diese variable bitte auch als instanzvariable erstellen und nicht nur lokal.

tobiasxschmidt commented on src/main/java/com/noxag/newnox/ui/configurationmodule/ConfigurationPane.java:
```diff
@@ -1,28 +1,71 @@
+        this.setBorder(new Border(
+                new BorderStroke(Color.BLACK, BorderStrokeStyle.SOLID, CornerRadii.EMPTY, BorderWidths.DEFAULT)));
+
+        // Add "run" button
```
den kompletten code der den run button erstellt in eigene methode auslagern, also
private void createRunButton (){...}
und dann createRunButton() hier aufrufen

tobiasxschmidt commented on src/main/java/com/noxag/newnox/ui/configurationmodule/ConfigurationPane.java:
```diff
@@ -1,28 +1,71 @@
+
+        fileChooser = new FileChooser();
+        fileChooser.setTitle("Open File...");
+        fileChooser.getExtensionFilters().addAll(new ExtensionFilter("PDF Files", "*.pdf"),
```
Schließt das hier alle dateiendungen außer .pdf aus? Oder kann man jetzt nur zusätzlich auch .pdfs auswählen?
Optimal wäre ersteres ;)

tobiasxschmidt commented on src/main/java/com/noxag/newnox/ui/configurationmodule/ConfigurationPane.java:
```diff
@@ -1,28 +1,71 @@
+        fileChooser = new FileChooser();
+        fileChooser.setTitle("Open File...");
+        fileChooser.getExtensionFilters().addAll(new ExtensionFilter("PDF Files", "*.pdf"),
+                new ExtensionFilter("All Files", "*.*"));
```
All Files wollen wir nicht, wirklich nur .pdf dateien

tobiasxschmidt commented on src/main/java/com/noxag/newnox/ui/configurationmodule/ConfigurationPane.java:
```diff
@@ -1,28 +1,71 @@
+    }
+
+    public Button getRunButton() {
+        // Gibt Run Button zurück
```
diesen Kommentar brauchen wir nicht, das ist offensichtlich ;)

tobiasxschmidt replied:
auch bei den anderen get* funktionen

luciehopp committed:
Instanzvariablen in private geändert
Erstellen der Buttons in createButtons() ausgelagert
Erstellen des FileChoosers in createFileChooser() ausgelagert
configTabPane zu Instanzvariable gemacht

PascalSchroederDE committed:
Added ActionEvent for Open-Button (File Chooser)

PascalSchroederDE committed:
Refactoring

PascalSchroederDE committed:
Prepared ConfigPage for adding checkboxes

luciehopp committed:
WIP

PascalSchroederDE committed:
Bug fixed; Refactored

PascalSchroederDE committed:
Prepared for merge
